lib/mirage_impl_random.ml

```diff
@@ -1,28 +1,25 @@
 open Functoria
-open Mirage_impl_misc
+open Mirage_impl_mclock
+open Mirage_impl_time
 
 type random = RANDOM
 let random = Type RANDOM
 
 let random_conf = object
   inherit base_configurable
-  method ty = random
+  method ty = time @-> mclock @-> random
   method name = "random"
-  method module_name = "Mirage_crypto_rng"
+  method module_name = "Mirage_crypto_rng_mirage.Make"
   method! keys = [ Mirage_key.(abstract prng) ]
   method! packages =
-    Mirage_key.(if_ is_unix)
-      [ package ~sublibs:["unix"] "mirage-crypto-rng" ]
-      [ package "mirage-crypto-entropy" ]
-  method! connect i _ _ =
-    match get_target i with
-    | #Mirage_key.mode_unix ->
-      "Lwt.return (Mirage_crypto_rng_unix.initialize ())"
-    | _ ->
-      (* here we could use the boot argument (--prng) to select the RNG! *)
-      "Mirage_crypto_entropy.initialize (module Mirage_crypto_rng.Fortuna)"
+    Mirage_key.pure [
+      package ~sublibs:["mirage"] ~min:"0.7.0" "mirage-crypto-rng"
+    ]
+  method! connect _i modname _ =
+    (* here we could use the boot argument (--prng) to select the RNG! *)
+    Fmt.strf "%s.initialize (module Mirage_crypto_rng.Fortuna)" modname
 end
 
-let default_random = impl random_conf
+let default_random = impl random_conf $ default_time $ default_monotonic_clock
```

Comment:
for those more into the mirage code base: I attempted
```OCaml
let default_random ?(time = default_time) ?(clock = default_monotonic_clock) () = impl random_conf $ time $ clock
```

which is what is done elsewhere as well, but there are some questions:
- who uses non-default labeled arguments? (if there isn't anyone, why bother?)
- on the use-site in other devices, we can no longer `?(random = default_random)`, but now require `time` and `mclock` explicitly and then `let random = match random with Some r -> r | None -> default_random ~time ~clock () in ...`
--> does, if there are now multiple calls to `default_random`, they all end up with the same device?
- since this is a minor release, I'd like to retain API compatibility, what is the supposed way to do so? expose a `Mirage.default_random_expert` with the new signature?